nengo/utils/tests/test_socket.py

```diff
@@ -0,0 +1,151 @@
+import math
+import threading
+import numpy as np
+
+import nengo
+from nengo.utils import sockets
+
+
+class SimThread(threading.Thread):
+    def __init__(self, simulator, sim_time, lock=None, wait=False):
+        threading.Thread.__init__(self)
+        self.sim = simulator
+        self.sim_time = sim_time
+        self.lock = lock
+        self.wait = wait
```

Comment:
Why is `self.wait` necessary? Does it ever get changed by any outside process?

Reply:
similarly, i think it's being just used to handle some startup issues, making sure that all three threads in the `test_send_recv_chain` are ready to go before any start simulating